draft-oid4vc-haip-sd-jwt-vc.md

```diff
@@ -128,18 +128,69 @@ Both sending Credential Offer same-device and cross-device is supported.
+   * The Wallet Attestation JWT scheme is defined in (#wallet-attestation-schema).
 
 Note: It is RECOMMENDED to use ephemeral client attestation JWTs for client authentication in order to prevent linkability across Credential Issuers.
 
 Note: Issuers should be mindful of how long the usage of the refresh token is allowed to refresh a credential, as opposed to starting the issuance flow from the beginning. For example, if the User is trying to refresh a credential more than a year after its original issuance, the usage of the refresh tokens is NOT RECOMMENDED.
 
 ### Wallet Attestation Schema {#wallet-attestation-schema}
 
-[Section 3.1 of wallet attestation draft would define the basics, and this profile will define the details.]
+Wallets MUST use attestations following the definition given in [@!I-D.ietf-oauth-attestation-based-client-auth].
+
+In addition to this definition, the Wallet Attestation MAY contain the following claims in the `cnf` element:
+
+* `key_type`: OPTIONAL. JSON String that asserts the security mechanism the Wallet uses to manage the private key associated with the public key given in the `cnf` claim. This mechanism is based on the capabilities of the execution environent of the Wallet, this might be a secure element (in case of a wallet residing on a smartphone) or a Cloud-HSM (in case of a cloud Wallet). This specification defines the following values for `key_type`:
+  * `software`: It MUST be used when the Wallet uses software-based key management.
+  * `hardware`: It MUST be used when the wallet uses hardware-based key management.
+  * `tee`: It SHOULD be used when the Wallet uses the Trusted Execution Environment for key management.
+  * `secure_enclave`: It SHOULD be used when the Wallet uses the Secure Enclave for key management.
+  * `strong_box`: It SHOULD be used when the Wallet uses the Strongbox for key management.
+  * `secure_element`: It SHOULD be used when the Wallet uses a Secure Element for key management.
+  * `hsm`: It SHOULD be used when the Wallet uses Hardware Security Module (HSM).
+* `user_authentication`: OPTIONAL. JSON String that asserts the security mechanism the Wallet uses to authenticate access to the private key associated with the public key given in the `cnf` claim. This specification defines the following values for `user_authentication`:
+  * `system_biometry`: It MUST be used when the key usage is authorized through a biometric factor by the operating system.
+  * `system_pin`: It MUST be used when the key usage is authorized through a knowledge factor by the operating system.
+  * `internal_biometry`: It MUST be used when the key usage is authorized through a biometric factor by the Wallet itself.
+  * `internal_pin`: It MUST be used when the key usage is authorized through a knowledge factor by the Wallet itself.
+  * `secure_element_pin` It MUST be used when the key usage is authorized through a knowledge factor by the secure element that is managing the key itself.
```

Comment:
```suggestion
* `user_authentication`: OPTIONAL. JSON String that asserts the security mechanism the Wallet uses to authenticate the user to authorize access to the private key associated with the public key given in the `cnf` claim. This specification defines the following values for `user_authentication`:
  * `system_biometry`: It MUST be used when the key usage is authorized by the mobile operating system using a biometric factor.
  * `system_pin`: It MUST be used when the key usage is authorized by the mobile operating system using personal identification number (PIN) as a knowledge factor.
  * `internal_biometry`: It MUST be used when the key usage is authorized by the Wallet using a biometric factor.
  * `internal_pin`: It MUST be used when the key usage is authorized by the Wallet using PIN as a knowledge factor.
  * `secure_element_pin` It MUST be used when the key usage is authorized by the secure element managing the key itself using PIN as a knowledge factor.
```

`knowledge factor` != `pin`, no? knowledge factor can be password, etc...
I have reservations by saying `by the Wallet`, given that the term `Wallet` is not defined in HAIP yet (opened #64), prefer something like `at the application layer`, but guess it's ok for now.

Reply:
I used knowledge factor because for in case of operating system, Android can not distinguish between password, pin and swipe pattern. Secure element will always have a pin and usually the wallet option as well

Reply:
what does that mean in terms of the current/proposed spec text?

Reply:
> I have reservations by saying by the Wallet
The section is about wallet attestation, so I think it is reasonable to use the term "Wallet"

Reply:
@Sakurann @paulbastian I suggest to just remove "as a knowledge factor" as I think PIN is sufficient.

Reply:
> I suggest to just remove "as a knowledge factor" as I think PIN is sufficient.

I edited the suggestion above - please apply it if you are ok with it

Reply:
> @Sakurann @paulbastian I suggest to just remove "as a knowledge factor" as I think PIN is sufficient.

sure, but the wallet needs to be defined - ideally in this PR, but for now I opened a separate issue #64

Reply:
looks fine to me